vmm/src/vm.rs

```diff
@@ -434,16 +439,22 @@ impl DeviceManager {
                 .map_err(DeviceManagerError::RegisterIoevent)?;
         }
 
-        // Assign IRQ to the virtio-pci device
-        let irqfd = EventFd::new(EFD_NONBLOCK).map_err(DeviceManagerError::EventFd)?;
-        let irq_num = allocator
-            .allocate_irq()
-            .ok_or(DeviceManagerError::AllocateIrq)?;
-        vm_fd
-            .register_irqfd(irqfd.as_raw_fd(), irq_num)
-            .map_err(DeviceManagerError::Irq)?;
-        // Let's use irq line INTA for now.
-        virtio_pci_device.assign_irq(irqfd, irq_num as u32, PciInterruptPin::IntA);
+        let vm_fd_clone = vm_fd.clone();
```

Comment:
It seems weird that we have to clone the VM fd and not take a reference on it. The lifetime of the IRQ closure will never exceeds that of the VM and its fd, it seems there should be a way of not cloning it and avoid the kvm-ioctl change.

Reply:
I used an `Arc<>` instead, works fine!